lib/models/NotificationConfiguration.js

```diff
@@ -0,0 +1,245 @@
+            }
+            const filterRules = filter[0].S3Key[0];
+            if (!filterRules.FilterRule || !filterRules.FilterRule[0]) {
+                filterObj.error = errors.MalformedXML.customizeDescription(
+                    'if included, queue configuration filter must contain a rule');
+                return filterObj;
+            }
+            const ruleArray = filterRules.FilterRule;
+            ruleArray.forEach(r => {
+                if (!r.Name || !r.Name[0] || !r.Value || !r.Value[0]) {
+                    filterObj.error = errors.MalformedXML.customizeDescription(
+                        'each included filter must contain a name and value');
+                } else if (!['Prefix', 'Suffix'].includes(r.Name[0])) {
+                    filterObj.error = errors.MalformedXML.customizeDescription(
+                        'filter Name must be one of Prefix or Suffix');
+                } else {
+                    filterObj.filterRules.push({
+                        name: r.Name[0],
+                        value: r.Value[0],
+                    });
+                }
+            });
+        }
+        return filterObj;
+    }
+
+    /**
+     * Check that id string is valid
+     * @param {string} id - id string (optional)
+     * @return {object} - contains error if parsing failed or id
+     */
+    _parseId(id) {
+        const idObj = {};
+        idObj.propName = 'ruleID';
+        if (id && id[0].length > 255) {
+            idObj.error = errors.InvalidArgument.customizeDescription(
+                'queue configuration ID is greater than 255 characters long');
+            return idObj;
+        }
+        if (!id || !id[0] || id[0] === '') {
+            // id is optional property, so create one if not provided or is ''
+            // We generate 48-character alphanumeric, unique id for rule
+            idObj.id = Buffer.from(UUID.v4()).toString('base64');
+        } else {
+            idObj.id = id[0];
+        }
+        // Each ID in a list of rules must be unique.
+        if (this._ids.includes(idObj.id)) {
+            idObj.error = errors.InvalidRequest.customizeDescription(
+                'queue configuration ID must be unique');
+            return idObj;
+        }
+        this._ids.push(idObj.id);
+        return idObj;
+    }
+
+    /**
+     * Check that arn string is valid
+     * @param {string} arn - queue arn
+     * @return {object} - contains error if parsing failed or queue arn
+     */
+    _parseArn(arn) {
+        const arnObj = {};
+        if (!arn || !arn[0]) {
+            arnObj.error = errors.MalformedXML.customizeDescription(
+                'each queue configuration must contain a queue arn');
+            return arnObj;
+        }
+        const splitArn = arn[0].split(':');
+        if (splitArn.length !== 6 ||
+            !(splitArn[0] === 'arn') ||
+            !(splitArn[1] === 'scality') ||
+            !(splitArn[2] === 'bucketnotif')) {
```

Comment:
Should use !== instead of !(===)
```suggestion
            splitArn[0] !== 'arn' ||
            splitArn[1] !== 'scality' ||
            splitArn[2] !== 'bucketnotif') {
```

Reply:
You could also slice and compare, something like
`splitArn.slice(3) == ['arn', 'scality', 'bucketnotif']`

Reply:
Yeah, I revamped this. After discussing with Naren, we don't need such individualized checks cos there's only one valid arn